src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java

```diff
@@ -36,21 +36,21 @@ public ResponseEntity<UserDTO> createUsers(@RequestBody UserDTO usersDto) {
   }
 
   @PutMapping(
-      value = "/{id}",
+      value = "/{emailId}",
       consumes = {MediaType.APPLICATION_JSON_VALUE},
       produces = {MediaType.APPLICATION_JSON_VALUE})
   public ResponseEntity<UserDTO> updateUsers(
-          @PathVariable("id") long id, @RequestBody UserDTO usersDto) {
-    UserDTO userDTO = getUserServicesAssembler().updateUsers(id, usersDto);
+          @PathVariable("emailId") String emailId, @RequestBody UserDTO usersDto) {
+    UserDTO userDTO = getUserServicesAssembler().updateUsers(emailId, usersDto);
     if (null == userDTO) {
-      throw new NotFoundException(MessageFormat.format("User by id {0} not found", id));
+      throw new NotFoundException(MessageFormat.format("User by id {0} not found", emailId));
     }
     return createResponse(userDTO, HttpStatus.OK);
   }
 
   @DeleteMapping("/{id}")
-  public ResponseEntity<HttpStatus> deleteUsers(@PathVariable("id") long id) {
-    getUserServicesAssembler().deleteUsers(id);
+  public ResponseEntity<HttpStatus> deleteUsers(@PathVariable("emailId") String emailId) {
```

Comment:
Email Id can have characters that can interfere with path. QueryParam would be a better option. Also, the Path id is still `id`  in the annotation :)